synthesizer/process/src/lib.rs

```diff
@@ -130,20 +173,50 @@ impl<N: Network> Process<N> {
     /// Adds a new stack to the process.
     /// If you intend to `execute` the program, use `deploy` and `finalize_deployment` instead.
     #[inline]
-    pub fn add_stack(&mut self, stack: Stack<N>) {
+    pub fn add_stack(&self, stack: Stack<N>) {
         // Add the stack to the process.
-        self.stacks.insert(*stack.program_id(), Arc::new(stack));
+        let mut lock = self.stacks.lock();
+        match lock {
+            Ok(ref mut lock) => lock.put(*stack.program_id(), Arc::new(stack)),
+            Err(_) => None,
+        };
     }
 }
 
 impl<N: Network> Process<N> {
     /// Initializes a new process.
     #[inline]
     pub fn load() -> Result<Self> {
+        // Assumption: this is only called in test code.
+        Process::load_from_storage(Some(aleo_std::StorageMode::Development(0)))
```

Comment:
Noting this may be the source of our troubles, and we may want to enclose the function in `#[cfg(any(test, feature = "test"))]` if possible.

Reply:
Yes this code needs to be updated.

Reply:
marking as draft again.

Reply:
[4aa125b](https://github.com/AleoNet/snarkVM/pull/2519/commits/4aa125bd3b9abef974d26c7f644307ffb221cd07) looks good.

If enclosing `fn load()` in something like `#[cfg(any(test, feature = "test"))]` is too difficult (perhaps because its called both by tests and benchmarks?), another safer approach could be to get rid of `fn load()` alltogether. Should take you just one search and replace to call `Process::load_from_storage(Some(aleo_std::StorageMode::Development(0)))` directly, and that keeps the `aleo_std::StorageMode::Development(0)` logic contained to within tests and benchmarks. That makes a future developer less likely to make mistakes.

Reply:
I realise also that the current approach can lead to deadlocks on the storage when running tests, because all the tests will try to access the same database... Looks like we also didn't run CI on this PR yet (you can ask @ljedrz for how he has been running CI with a separate -ci branch).

New proposal: what about we let `fn load` use `store: None`. Just like `load_from_storage `, it should load the `credits.aleo` program because the tests need it. To avoid adding too many lines of code, you can still abstract similar logic to a shared function.

Reply:
We can not use `None` as this breaks many tests.

Reply:
Clear. Then we should rename `load` to `load_testing_only`.